tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,650 @@
+lazy_static::lazy_static! {
+    // When iterating recursively, this keeps track of the current path
+    static ref CURRENT_PATH: Mutex<Vec<TypeAndField>> = Mutex::new(vec![]);
+    // a set of CURRENT_PATH whenever it hits a pii=true field
+    static ref PII_TYPES: Mutex<BTreeSet<Vec<TypeAndField>>> = Mutex::new(BTreeSet::new());
+    // All the structs and enums, where the key is the full path. The string in the tuple represents
+    // the module path to the item. Now that I think of it, should be able to just get it from the key.
+    static ref ALL_TYPES: Mutex<HashMap<String, EnumOrStruct>> = Mutex::new(HashMap::new());
+    /// The values that we wanna match on, (true, false, maybe)
+    static ref PII_VALUES: Mutex<Vec<String>> = Mutex::new(vec![]);
+    /// All the use statements in a given module.
+    static ref USE_STATEMENTS: Mutex<BTreeMap<String, BTreeSet<String>>> = Mutex::new(BTreeMap::new());
+}
+
+/// It's only purpose is to populate ALL_TYPES and USE_STATEMENTS
+struct FileSyntaxVisitor {
+    module_path: String,
+}
+
+impl FileSyntaxVisitor {
+    fn insert_use_statements(&self, use_statements: Vec<String>) {
+        USE_STATEMENTS
+            .lock()
+            .unwrap()
+            .entry(self.module_path.clone())
+            .or_default()
+            .extend(use_statements);
+    }
+
+    fn fill_pii_hashmaps(paths: &Vec<PathBuf>) {
+        for path in paths {
+            // first iterate through all rust files to create the hashmap
+            let mut hashmap_filler = FileSyntaxVisitor {
+                module_path: rust_file_to_use_path(path.as_path()),
+            };
+            let file_content = fs::read_to_string(path.as_path()).unwrap();
+            let syntax_tree: syn::File = syn::parse_file(&file_content).unwrap();
+            hashmap_filler.visit_file(&syntax_tree);
+        }
+    }
+}
+
+impl<'ast> Visit<'ast> for FileSyntaxVisitor {
+    fn visit_item_struct(&mut self, node: &'ast ItemStruct) {
+        // These are pretty common and we know they dont contain PII so might as well not include
+        // them to save time.
+        for common_value in ["Annotated", "Value"] {
+            if node.ident == common_value {
+                return;
+            }
+        }
+
+        let struct_name = format!("{}::{}", self.module_path, node.ident);
+        self.insert_use_statements(vec![struct_name.clone()]);
+        ALL_TYPES
+            .lock()
+            .unwrap()
+            .insert(struct_name, EnumOrStruct::Struct(node.clone()));
+    }
+
+    fn visit_item_enum(&mut self, node: &'ast ItemEnum) {
+        let enum_name = format!("{}::{}", self.module_path, node.ident);
+        self.insert_use_statements(vec![enum_name.clone()]);
+        ALL_TYPES
+            .lock()
+            .unwrap()
+            .insert(enum_name, EnumOrStruct::Enum(node.clone()));
+    }
+
+    fn visit_item_use(&mut self, i: &'ast syn::ItemUse) {
```

Comment:
```suggestion
    fn visit_use_statement(&mut self, i: &'ast syn::ItemUse) {
```
Not very sure about this.